fuzz/lnwire/channelannouncement/channel_announcement.go

```diff
@@ -0,0 +1,21 @@
+// +build gofuzz
+
+package channelannouncement
+
+import (
+	fuzz "github.com/lightningnetwork/lnd/fuzz/lnwire"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// Fuzz is used by go-fuzz.
+func Fuzz(data []byte) int {
```

Comment:
High level question: why do certain message targets like `AcceptChannel` use a custom scenario rather than the `fuzz.Harness` scenario as many of these below do?

Reply:
Yeah so this could have been more explicit. it's because of parsing UpfrontShutdownScript - before the test starts, if it's empty (nil slice), it will be converted into a non-nil slice at the very end of the test (the decoding+encoding changes it). since there's no way to only reflect.DeepEqual subsets of a struct (or like switch on error / determine which field the error occured), had to do it this way